db/compaction/compaction_job.cc

```diff
@@ -295,10 +300,16 @@ void CompactionJob::Prepare() {
       ROCKS_LOG_WARN(db_options_.info_log,
                      "Failed to get current time in compaction: Status: %s",
                      status.ToString().c_str());
-      penultimate_level_cutoff_seqno_ = 0;
+      // preserve all time information
+      preserve_time_min_seqno_ = 0;
     } else {
-      penultimate_level_cutoff_seqno_ =
-          seqno_time_mapping_.TruncateOldEntries(_current_time);
+      seqno_time_mapping_.TruncateOldEntries(_current_time);
+      uint64_t preserve_time =
+          static_cast<uint64_t>(_current_time) > preserve_time_duration
+              ? _current_time - preserve_time_duration
+              : 0;
+      preserve_time_min_seqno_ =
+          seqno_time_mapping_.GetOldestSequenceNum(preserve_time);
```

Comment:
I didn't completely get why now the oldest entry isn't seqno to preserve anymore, but at least it's still correct.

Reply:
Yeah, you're right, it is the oldest entry for this version of code. (my first version was having preserve_time and preclude_time differently, then I changed mind to have preclude_time to override preserve_time if both set, but forgot to revert this code. Now I think we're going back to have them differently).